symengine/sets.h

```diff
@@ -161,12 +181,23 @@ inline RCP<const UniversalSet> universalset()
 }
 
 //! \return RCP<const Set>
+inline RCP<const Set> finiteset(const set_basic &container)
+{
+    if (FiniteSet::is_canonical(container)) {
+        return make_rcp<const FiniteSet>(container);
+    }
+    return emptyset();
+}
+
+//! \return RCP<const Set>
 inline RCP<const Set> interval(const RCP<const Number> &start,
                                const RCP<const Number> &end,
                                const bool left_open = false,
                                const bool right_open = false)
 {
     if (Interval::is_canonical(start, end, left_open, right_open)) {
+        if (eq(*start, *end) and not(left_open or right_open))
+            return finiteset({start});
```

Comment:
+1. Shouldn't this be in `Interval::is_canonical` as well? The change will complicate the logic though